src/mitol/payment_gateway/api.py

```diff
@@ -361,6 +433,106 @@ def prepare_checkout(
+            if settings.ENVIRONMENT == "prod"
+            else CYBERSOURCE_TEST_API_URL,
+            "merchant_keyid": settings.MITOL_PAYMENT_GATEWAY_CYBERSOURCE_MERCHANT_SECRET_KEY_ID,
+            "merchant_secretkey": settings.MITOL_PAYMENT_GATEWAY_CYBERSOURCE_MERCHANT_SECRET,
+            "timeout": 1000,
+        }
+        return configuration_dictionary
+
+    def perform_refund(self, refund):
+        """
+        Based on the provided refund detail data object, this method calls the cyber source refund API call after
+        performing the appropriate operation on the data.
+
+        returns:
+
+            i) API Success:
+                Returns a ProcessorResponse object that the caller app can use to perform whatever they need
+
+            ii)  API Failure ("DUPLICATE_REQUEST"):
+                Raises RefundDuplicateException when the refund API call is duplicate
+
+            iii)  API Failure (General):
+                Raises the exception(Mostly CyberSource.rest.ApiException) to the caller to decide what their
+                application behaviour
+        """
+
+        api_instance = RefundApi(self.get_client_configuration())
+        refund_payload = self.generate_refund_payload(refund)
+        transaction_id = refund.transaction_id
+
+        try:
+            return_data, status, body = api_instance.refund_payment(
+                refund_payload, transaction_id
+            )
+            _body = json.loads(body)
+            # Transforming the response in a format that process response decoder can work on while keeping our
+            # structure consistent
+            _response_transformed = {
+                "data": _body,
+                "reason_code": _body.get("processorInformation", {}).get(
+                    "responseCode", ""
+                ),
+                "transaction_id": transaction_id,
+                "message": "The refund request was successful",
+                "decision": _body["status"],
+            }
+
+            response = self.decode_processor_api_response(_response_transformed)
+            return response
+
+        except Exception as ex:
+            _body = json.loads(ex.body)
+
+            # Special case for request failure when DUPLICATE_REQUEST
+            if _body["reason"] == ProcessorResponse.STATE_DUPLICATE:
+                raise RefundDuplicateException(
+                    _body["reason"],
+                    transaction_id,
+                    refund.refund_amount,
+                    _body,
+                    message=_body["message"],
+                )
+
+            raise ex
+
+    def generate_refund_payload(self, refund):
+        """
+        CyberSource API client would expect the payload in a specific format to perform the refund API call.
+        This method generated that payload out of the Refund data class(s)
+        """
+        transaction_id = refund.transaction_id
+        orderInformationAmountDetails = (
```

Comment:
This and the other variables here should be `snake_case`.

Reply:
Thanks for pointing to this. I definitely overlooked it since I was porting over the data model creation from the CyberSource Samples client while initially testing things out and these somehow remained the same(🤯 ). Should be fixed when I address the feedback.